### PR TITLE
xbutil2: Adding xclbin uuid to CU report 

### PR DESCRIPTION
**Pull request by AShivangi**
- **CR-1100427** ASTeR basic sanity - xbmgmt2 examine - ERROR: Failed to open /sys/bus/pci/devices/0000:d8:00.0//xclbinuuid for reading
- **CR-1083710** display xclbin uuid info in xbutil examine command (re-do)

We have to move xclbin-uuid to compute units because:
1. xclbinuuid is not available for xbmgmt
2. we want to show the uuid only for dynamic-regions report (the earlier change had it displayed for all reports)

Sample outputs:
```
$ xbutil examine -d 17:00  -r compute-units -f json -o o.txt --force
----------------------------------------------
1/1 [0000:17:00.1] : xilinx_u30_gen3x4_base_1 
----------------------------------------------
Xclbin UUID                                   
  6250EC80-3F38-4BE0-AC90-68BFD3C140A8        

Compute Units
  PL Compute Units
    Index   Name                          Base_Address    Usage   Status  
    0       verify:verify_1               0x1400000       1       (IDLE)  

  PS Compute Units
    Index   Name                          Base_Address    Usage   Status  

Successfully wrote the json file: o.txt

$cat o.txt
{                                      
    "schema_version": {                
        "schema": "JSON",              
        "creation_date": "Thu May 20 21:33:04 2021 GMT"
    },                                                 
    "devices": [                                       
        {                                              
            "interface_type": "pcie",                  
            "device_id": "0000:17:00.1",               
            "dynamic_regions": [                       
                {                                      
                    "xclbin_uuid": "6250EC80-3F38-4BE0-AC90-68BFD3C140A8",
                    "compute_units": [                                    
                        {                                                 
                            "name": "verify:verify_1",                    
                            "base_address": "0x1400000",                  
                            "usage": "1",                                 
                            "type": "PL",                                 
                            "status": {                                   
                                "bit_mask": "0x4",                        
                                "bits_set": [                             
                                    "IDLE"                                
                                ]                                         
                            }                                             
                        }                                                 
                    ]                                                     
                }
            ]
        }
    ]
}
```

### CODE REVIEW

File: src/runtime_src/core/tools/common/ReportCu.cpp [hunk 1]
```diff
@@ -129,7 +129,14 @@ populate_cus(const xrt_core::device *device)
...
   }
 
-  return pt;
+  boost::property_tree::ptree ptree;
+  try {
+    std::string uuid = xrt::uuid(xrt_core::device_query<xrt_core::query::xclbin_uuid>(device)).to_string();
+    boost::algorithm::to_upper(uuid);
+    ptree.put("xclbin_uuid", uuid);
+  } catch (...) {  }
+  ptree.add_child("compute_units", pt);
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportCu.cpp:138]: Why is there another node called "compute_units" in the JSON tree.  This looks wrong.

**Review by rozumx — CHANGES_REQUESTED** [review]
The JSON schema changed...and looks wrong.

Please do reach out to me so I can further understand why this was needed.

File: src/runtime_src/core/tools/common/ReportCu.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/common/ReportCu.cpp:138]: Thanks for the chat. I have made the changes that we dicscussed

**Comment by stsoe** [comment]
clang-tidy review says "All clean, LGTM! :+1:"

**Comment by AShivangi** [comment]
retest this please

File: src/runtime_src/core/tools/common/ReportDynamicRegion.cpp [hunk 2]
```diff
@@ -268,49 +284,57 @@ ReportCu::writeReport( const xrt_core::device* /*_pDevice*/,
...
+    _output << "  " + dfx.get<std::string>("xclbin_uuid", "N/A") << std::endl;
+    _output << std::endl;
+
+    const boost::property_tree::ptree& pt_cu = dfx.get_child("compute_units", empty_ptree);
+    _output << "Compute Units" << std::endl;
+    _output << "  PL Compute Units" << std::endl;
+    _output << cuFmt % "Index" % "Name" % "Base_Address" % "Usage" % "Status";
+    try {
+      int index = 0;
+      for(auto& kv : pt_cu) {
```
Comment (rozumx) [inline src/runtime_src/core/tools/common/ReportDynamicRegion.cpp:303]: This for loop is for ranges and yet the code wishes to treat it as an array.  For future implementations, you might wish to consider the following:
```
for (std::size_t index = 0; index < pt_cu.size(); ++index) {
  auto & kv = pt_cu[index];
...
```
Makes the intent and code a little easier to read.

And yes, I know it goes against the coding guidelines: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Res-for-range

Side Note
If we were using C++20, then we could potentially use:
`for( unsigned int count = 0; auto& kv : pt_cu)`

**Review by rozumx — APPROVED** [review]
Approve changes.

**Pull request merged**